File: blSystem-frontend/src/app/model/Donor.ts
import { BloodGroup } from "./enums/BloodGroup";
import { Gender } from "./enums/Gender";

export class Donor {
    id!: number;
    name!: string;
    gender!: Gender;
    age!: number;
    address!: string;
    bloodGroup!: BloodGroup;
    contactNumber!: string;
    lastDonationDate! : Date;
    eligible?: boolean 

    constructor() {}
  }
  

File: blSystem-backend/src/main/java/com/blsystem/entity/Donor.java
package com.blsystem.entity;

import com.blsystem.entity.enums.BloodGroup;
import com.blsystem.entity.enums.Gender;
import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Table(name = "donors")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Donor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;
        private String name;
        private Gender gender;
        private int age;
        private String address;
        private BloodGroup bloodGroup;
        private String contactNumber;
        @JsonFormat(pattern = "yyyy-MM-dd")
        private Date lastDonationDate;
        private boolean isEligible;

}
